### ISSUE TITLE
Magit 3.0 (and 3.1.1) open COMMIT_EDITMSG in a new frame

### ISSUE THREAD

**Issue by nrnrnr** [magit/magit#4434]
(I'm aware of previous issue https://github.com/magit/magit/issues/2172 on this topic, but it seemed best to open a new issue.)

I'm using these versions:

    Magit 3.0.0, Git 2.20.1, Emacs 26.1, gnu/linux

 1.  Behavior I expected: On typing `c c` in a Magit window, I expected Magit to use `display-buffer` to open `COMMIT_EDITMSG` in the other window.
 2. Behavior I observer: Magit opens a window in a new frame.  The frame is split into two windows.  The upper window shows `COMMIT_EDITMSG`,  and the lower window is a `magit-diff: $DIRNAME` window.
 3. To reproduce the issue, create a file `magit-only.el` containing this elisp code:
     ```
        (package-initialize)
        (require 'magit)
    ```
    Then run `emacs --no-init-file -l ~/.emacs.d/magit-only.el`.\
    Then go to a git directory, type `C-x g`, make sure there is something staged in the index, and type `c c`.

There is no error, so I don't have a backtrace.

This thread on Reddit may be relevant: https://www.reddit.com/r/emacs/comments/hllud8/how_can_one_configure_magit_to_not_open_a_new/

----

As an alternative way of loading only Magit and its dependencies, I ran

    M-x magit-emacs-Q-command RET

which produced this shell command:

````
/usr/bin/emacs -Q --eval \(setq\ debug-on-error\ t\) -L /home/nr/.emacs.d/elpa/dash-20210609.1330/ -L /home/nr/.emacs.d/elpa/transient-0.3.5/ -L /home/nr/.emacs.d/elpa/with-editor-20210524.1654/ -L /home/nr/.emacs.d/elpa/magit-3.0.0/ -L /home/nr/.emacs.d/elpa/git-commit-20210525.844/ -l /home/nr/.emacs.d/elpa/magit-3.0.0/magit
````

Running this command resulted in the following error:

````
Debugger entered--Lisp error: (file-missing "Cannot open load file" "No such file or directory" "magit-section")
  require(magit-section)
  byte-code("\301\302!\210\301\303!\210\301\304!\210\301\305!\210\301\306!\210\301\307!\210\301\310!\210\301\311!\210\312 \203=\0\31310\0\301\314!0\202<\0\315\020\316\317\002\"\210\316\320!\262\001\210\321\322\323\324\325\326\325\327\325\330\331\332&\013\210\321\333\323\334\325\335\331\322&\007\210\321\336\323\337\331\322%\210\321\340\323\341\331\322%\210\321\342\323\343\331\322%\210\321\344\323\345\325\346\331\322&\007\210\321\347\323\350\325\351\331\322\331\344&\011\210\321\352\323\353\331\322\331\354&\007\210\321\355\323\356\331\322%\210\357\342\360\361#\210\357\352\362\361#\210\357\342\363\361#\210\357\352\364\361#\210\357\322\303\361#\210\357\352\365\361#\210\357\307\366\361#\210\321\367\323\370\325\371\325\372\325\373\325\374\331\322\331\355\331\333&\021\210\357\367\375\376#\210\357\377\375\376#\210\357\367\201@\0\376#\210\357\367\201A\0\376#\210\357\367\201B\0\376#\210\201C\0\201D\0!\207" [magit-inhibit-libgit require magit-utils magit-section magit-git magit-mode magit-margin magit-process magit-transient magit-autorevert magit--libgit-available-p (error) magit-libgit error message "Error while loading `magit-libgit': %S" "That is not fatal.  The `libegit2' module just won't be used." custom-declare-group magit nil "Controlling Git from Emacs." :link (url-link "https://magit.vc") (info-link "(magit)FAQ") (info-link "(magit)") :group tools magit-essentials "Options that every Magit user should briefly think about.\n\nEach of these options falls into one or more of these categories:\n\n* Options that affect Magit's behavior in fundamental ways.\n* Options that affect safety.\n* Options that affect performance.\n* Options that are of a personal nature." (info-link "(magit)Essential Settings") magit-miscellaneous "Miscellaneous Magit options." magit-commands "Options controlling behavior of certain commands." magit-modes "Modes used or provided by Magit." magit-buffers "Options concerning Magit buffers." (info-link "(magit)Modes and Buffers") magit-refresh "Options controlling how Magit buffers are refreshed." (info-link "(magit)Automatic Refreshing of Magit Buffers") magit-faces "Faces used by Magit." faces magit-extensions "Extensions to Magit." custom-add-to-group git-commit custom-group git-commit-faces git-rebase git-rebase-faces magit-section-faces with-editor magit-related "Options that are relevant to Magit but that are defined elsewhere." (custom-group-link vc) (custom-group-link smerge) (custom-group-link ediff) (custom-group-link auto-revert) auto-revert-check-vc-info custom-variable magit-auto-revert ediff-window-setup-function smerge-refine-ignore-whitespace vc-follow-symlinks provide magit-core] 18)
  require(magit-core)
  byte-code("\300\301!\210\300\302!\210\300\303!\210\300\304!\210\300\305!\210\300\306!\210\300\307!\210\300\310!\210\300\311\312\313#\210\300\314!\207" [require magit-core magit-diff magit-log magit-wip magit-apply magit-repos git-commit format-spec package nil t with-editor] 4)
  load("/home/nr/.emacs.d/elpa/magit-3.0.0/magit" nil t)
  command-line-1(("--eval" "(setq debug-on-error t)" "-L" "/home/nr/.emacs.d/elpa/dash-20210609.1330/" "-L" "/home/nr/.emacs.d/elpa/transient-0.3.5/" "-L" "/home/nr/.emacs.d/elpa/with-editor-20210524.1654/" "-L" "/home/nr/.emacs.d/elpa/magit-3.0.0/" "-L" "/home/nr/.emacs.d/elpa/git-commit-20210525.844/" "-l" "/home/nr/.emacs.d/elpa/magit-3.0.0/magit"))
  command-line()
  normal-top-level()
````

I have confirmed that the "various Ns" in the commands are the correct versions---or at least that those versions exist in my filesystem.  I have also noted that emacs wanted `git-commit` not `ghub`.

**Comment by tarsius** [comment]
The `magit-emacs-Q-command` issue was fixed right after releasing 3.0.0 in f50c3aa56704cddd46ca987722642584bc8290b0. Please update and try again.

**Comment by nrnrnr** [comment]
I've upgraded Magit and its dependencies using the `U` command on the package menu.  Exited Emacs and restarted (so hopefully no frankenpackage).  It now reports

```
Magit 3.1.1, Git 2.20.1, Emacs 26.1, gnu/linux
```

Running `magit-emacs-Q-command` produces

```
/usr/bin/emacs -Q --eval \(setq\ debug-on-error\ t\) -L /home/nr/.emacs.d/elpa/dash-20210609.1330/ -L /home/nr/.emacs.d/elpa/transient-0.3.6/ -L /home/nr/.emacs.d/elpa/with-editor-20210524.1654/ -L /home/nr/.emacs.d/elpa/magit-3.1.1/ -L /home/nr/.emacs.d/elpa/magit-section-3.1.1/ -L /home/nr/.emacs.d/elpa/git-commit-20210525.844/ -l /home/nr/.emacs.d/elpa/magit-3.1.1/magit
```

Version numbers all appear to be correct for my filesystem.

Starting Emacs with this command results in a stack trace, with `*Backtrace*` buffer as follows:

```
Debugger entered--Lisp error: (wrong-type-argument stringp nil)
  call-process("/usr/bin/movemail" nil #<buffer  *temp*> nil "--version")
  rmail-probe("/usr/bin/movemail")
  rmail-autodetect()
  rmail-movemail-variant-p()
  byte-code("\300 \210\301\302\303\304\305DD\306\307\310\311\312\313\314&\011\210\315\316\317\320#\210\321\211\203<\0\211@\316\001N\2035\0\317\001N\2045\0\322\317\002\316\004N#\210\001A\266\202\202\033\0\210\323\316\317\324#\207" [rmail-movemail-variant-p custom-declare-variable rmail-user-mail-address-regexp funcall function #f(compiled-function () #<bytecode 0x101b175>) "Regexp matching user mail addresses.\nIf non-nil, this variable is used to identify the correspondent\nwhen receiving new mail.  If it matches the address of the sender,\nthe recipient is taken as correspondent of a mail.\nIf nil (default value), your `user-login-name' and `user-mail-address'\nare used to exclude yourself as correspondent.\n\nUsually you don't have to set this variable, except if you collect mails\nsent by you under different user names.\nThen it should be a regexp matching your mail addresses.\n\nSetting this variable has an effect only before reading a mail." :type (choice (const :tag "None" nil) regexp) :group rmail-retrieve :version "21.1" defvaralias rmail-dont-reply-to-names mail-dont-reply-to-names nil (saved-value saved-variable-comment) put make-obsolete-variable "24.1"] 10)
  require(rmail)
  byte-code("\300\301!\210\302\303\304\"\207" [require rmail autoload rmail-update-summary "rmailsum"] 3)
  require(gnus-util)
  byte-code("\300\301!\210\300\302!\210\300\303!\210\300\304!\210\300\305!\210\306\307\310\"\210\306\311\312\"\210\306\313\312\"\210\306\314\315\"\210\306\316\315\"\210\306\317\320\"\210\306\321\320\"\210\306\322\320\"\210\306\323\324\"\210\306\325\326\"\210\306\327\326\"\210\306\330\326\"\210\306\331\326\"\210\306\332\326\"\210\306\333\326\"\210\306\334\335\"\210\306\322\320\"\210\306\336\320\"\207" [require gnus-util epg epa password-cache mm-encode autoload mail-strip-quoted-names "mail-utils" mml2015-sign "mml2015" mml2015-encrypt mml1991-sign "mml1991" mml1991-encrypt message-fetch-field "message" message-goto-body message-options-get mml-insert-tag "mml" mml-smime-sign "mml-smime" mml-smime-encrypt mml-smime-sign-query mml-smime-encrypt-query mml-smime-verify mml-smime-verify-test epa--select-keys "epa" message-options-set] 3)
  require(mml-sec)
  byte-code("\301\302!\210\301\303!\210\301\304!\210\301\305!\210\301\306!\210\307\310\311\"\210\307\312\313\"\210\307\314\311\"\210\307\315\311\"\210\307\316\311\"\210\307\317\320\"\210\307\321\311\"\210\307\322\323\"\210\307\324\311\"\210\307\325\311\"\210\307\326\311\"\210\307\327\311\"\210\330\331\332\333\334\335\336\337\340\341&\011\210\330\342\343\344\334\335\336\345\340\341&\011\210\330\346\347\350\334\351\336\352\030\353\354\355\356\357\360\361\362\363\364\365\366\010\257\007\361\362\367\370\355\371\353\357\372\354\355\364\365\361\357\373\364\374\375\010\257\007\257\010\257\007\361\362\376\364\365\377\010\257\007\257\006F)\340\341&\011\210\330\201@\0\201A\0\201B\0\334\201C\0\336\201D\0\340\341&\011\210\330\201E\0\201A\0\201F\0\334\201C\0\336\201D\0\340\341&\011\207" [dispositions require mm-util mm-bodies mm-encode mm-decode mml-sec autoload message-make-message-id "message" gnus-completing-read "gnus-util" message-fetch-field message-mark-active-p message-info fill-flowed-encode "flow-fill" message-posting-charset dnd-get-local-file-name "dnd" message-options-set message-narrow-to-head message-in-body-p message-mail-p custom-declare-variable mml-content-type-parameters '(name access-type expiration size permission format) "A list of acceptable parameters in MML tag.\nThese parameters are generated in Content-Type header if exists." :version "22.1" :type (repeat (symbol :tag "Parameter")) :group message mml-content-disposition-parameters '(filename creation-date modification-date read-date) "A list of acceptable parameters in MML tag.\nThese parameters are generated in Content-Disposition header if exists." (repeat (symbol :tag "Parameter")) mml-content-disposition-alist '((text (rtf . "attachment") (t . "inline")) (t . "attachment")) "Alist of MIME types or regexps matching file names and default dispositions.\nEach element should be one of the following three forms:\n\n  (REGEXP . DISPOSITION)\n  (SUPERTYPE (SUBTYPE . DISPOSITION) (SUBTYPE . DISPOSITION)...)\n  (TYPE . DISPOSITION)\n\nWhere REGEXP is a string which matches the file name (if any) of an\nattachment, SUPERTYPE, SUBTYPE and TYPE should be symbols which are a\nMIME supertype (e.g., text), a MIME subtype (e.g., plain) and a MIME\ntype (e.g., text/plain) respectively, and DISPOSITION should be either\nthe string \"attachment\" or the string \"inline\".  The value t for\nSUPERTYPE, SUBTYPE or TYPE matches any of those types.  The first\nmatch found will be used." "23.1" (radio :format "DISPOSITION: %v" :value "attachment" (const :format "%v " "attachment") (const :format "%v\n" "inline")) repeat :offset 0 choice :format "%[Value Menu%]%v" cons :tag "(REGEXP . DISPOSITION)" :extra-offset 4 (regexp :tag "REGEXP" :value ".*") "(SUPERTYPE (SUBTYPE . DISPOSITION)...)" :indent (symbol :tag "    SUPERTYPE" :value text) "%v%i\n" "%v" 5 (symbol :tag "SUBTYPE" :value t) "(TYPE . DISPOSITION)" (symbol :tag "TYPE" :value t) mml-insert-mime-headers-always t "If non-nil, always put Content-Type: text/plain at top of empty parts.\nIt is necessary to work against a bug in certain clients." "24.1" boolean mml-enable-flowed "If non-nil, enable format=flowed usage when encoding a message.\nThis is only performed when filling on text/plain with hard\nnewlines in the text."] 34)
  require(mml)
  byte-code("\300\301!\210\300\302!\210\300\303!\210\300\304!\210\300\305!\210\300\306!\210\300\307!\210\300\310!\210\300\311!\210\300\312!\210\300\313!\210\300\314!\210\300\315!\210\316\317\320\"\210\321\322\323\324\325\326\327\330\327\331&\011\210\321\332\333\334\325\335\327\322&\007\210\321\336\333\337\325\340\327\322&\007\210\321\341\333\342\325\343\327\322&\007\210\321\344\333\345\325\346\327\322&\007\210\321\347\333\350\325\351\327\344&\007\210\321\352\333\353\325\354\327\322&\007\210\321\355\333\356\325\357\327\322&\007\210\321\360\333\361\327\322%\210\321\362\333\363\327\322%\210\321\364\333\365\327\322\327\366&\007\210\367\370\371\372\373DD\374\327\332\375\376&\007\210\367\377\371\372\201@\0\ DD\201A\0\327\336\375\201B\0&\007\210\367\201C\0\371\372\201D\0\ DD\201E\0\327\336\375\201F\0&\007\210\367\201G\0\371\372\201H\0\ DD\201I\0\375\201J\0\327\341&\007\210\367\201K\0\371\372\201L\0\ DD\201M\0\201N\0\201O\0\375\201P\0\327\341&\011\210\367\201Q\0\371\372\201R\0\ DD\201S\0\327\341\375\201T\0&\007\210\367\201U\0\371\372\201V\0\ DD\201W\0\327\344\375\201X\0&\007\210\367\201Y\0\371\372\201Z\0\ DD\201[\0\201N\0\201\\\0\375\201]\0\327\355&\011\210\367\201^\0\371\372\201_\0\ DD\201`\0\201N\0\201O\0\327\355\375\201P\0&\011\210\367\201a\0\371\372\201b\0\ DD\201c\0\327\360\375\201d\0&\007\210\367\201e\0\371\372\201f\0\ DD\201g\0\201N\0\201O\0\327\360\327\355\325\201h\0\375\201i\0&\015\210\367\201j\0\371\372\201k\0\ DD\201l\0\201N\0\201O\0\327\360\327\355\325\201m\0\375\201n\0&\015\210\367\201o\0\371\372\201p\0\ DD\201q\0\327\360\327\355\325\201r\0\375\201s\0&\013\210\367\201t\0\371\372\201u\0\ DD\201v\0\327\362\327\355\325\201w\0\375\201x\0&\013\210\367\201y\0\371\372\201z\0\ DD\201{\0\201N\0\201|\0\327\362\327\355\325\201}\0\375\201~\0&\015\210\367\201\177\0\371\372\201\200\0\ DD\201\201\0\327\355\325\201\202\0\375\201\203\0&\011\210\367\201\204\0\371\372\201\205\0\ DD\201\206\0\327\360\327\355\325\201\207\0\375\201\210\0&\013\210\367\201\211\0\371\372\201\212\0\ DD\201\213\0\327\362\327\355\325\201\214\0\375\201X\0&\013\210\367\201\215\0\371\372\201\216\0\ DD\201\217\0\327\344\325\201\220\0\375\201\221\0&\011\210\367\201\222\0\371\372\201\223\0\ DD\201\224\0\327\332\325\201\225\0\375\201X\0&\011\210\367\201\226\0\371\372\201\227\0\ DD\201\230\0\201N\0\201|\0\375\201\231\0\325\201\232\0\327\332&\013\210\367\201\233\0\371\372\201\234\0\ DD\201\235\0\201N\0\201O\0\327\332\325\201\236\0\375\201X\0&\013\210\367\201\237\0\371\372\201\240\0\ DD\201\241\0\201N\0\201O\0\327\332\325\201\242\0\375\201X\0&\013\210\367\201\243\0\371\372\201\244\0\ DD\201\245\0\201N\0\201O\0\375\201\246\0\325\201\247\0\327\332&\013\210\367\201\250\0\371\372\201\251\0\ DD\201\252\0\201N\0\201O\0\375\201\246\0\325\201\253\0\327\332&\013\210\367\201\254\0\371\372\201\255\0\ DD\201\256\0\201N\0\201O\0\375\201\246\0\325\201\257\0\327\332&\013\210\367\201\260\0\371\372\201\261\0\ DD\201\262\0\201N\0\201O\0\375\201\263\0\325\201\264\0\327\332&\013\207" [require mailheader gmm-utils mail-utils mailabbrev mail-parse mml rfc822 format-spec dired mm-util rfc2047 puny rmc autoload mailclient-send-it "mailclient" custom-declare-group message ((user-mail-address custom-variable) (user-full-name custom-variable)) "Mail and news message composing." :link (custom-manual "(message)Top") :group mail news message-various nil "Various Message Variables." (custom-manual "(message)Various Message Variables") message-buffers "Message Buffers." (custom-manual "(message)Message Buffers") message-sending "Message Sending." (custom-manual "(message)Sending Variables") message-interface "Message Interface." (custom-manual "(message)Interface") message-forwarding "Message Forwarding." (custom-manual "(message)Forwarding") message-insertion "Message Insertion." (custom-manual "(message)Insertion") message-headers "Message Headers." (custom-manual "(message)Message Headers") message-news "Composing News Messages." message-mail "Composing Mail Messages." message-faces "Faces used for message composing." faces custom-declare-variable message-directory funcall function #f(compiled-function () #<bytecode 0xfe597d>) "Directory from which all other mail file variables are derived." :type directory message-max-buffers #f(compiled-function () #<bytecode 0xfe5989>) "How many buffers to keep before starting to kill them off." integer message-send-rename-function #f(compiled-function () #<bytecode 0xfe5995>) "Function called to rename the buffer after sending it." (choice function (const nil)) message-fcc-handler-function #f(compiled-function () #<bytecode 0xfe59a1>) "A function called to save outgoing articles.\nThis function will be called with the name of the file to store the\narticle in.  The default function is `message-output' which saves in Unix\nmailbox format." (radio (function-item message-output) (function :tag "Other")) message-fcc-externalize-attachments #f(compiled-function () #<bytecode 0xfe59ad>) "If non-nil, attachments are included as external parts in Fcc copies." :version "22.1" boolean message-courtesy-message #f(compiled-function () #<bytecode 0xfe59b9>) "This is inserted at the start of a mailed copy of a posted message.\nIf the string contains the format spec \"%s\", the Newsgroups\nthe article has been posted to will be inserted there.\nIf this variable is nil, no such courtesy message will be added." (radio string (const nil)) message-ignored-bounced-headers #f(compiled-function () #<bytecode 0xfe59c5>) "Regexp that matches headers to be removed in resent bounced mail." regexp message-from-style #f(compiled-function () #<bytecode 0xfe59d1>) "Specifies how \"From\" headers look.\n\nIf nil, they contain just the return address like:\n\011king@grassland.com\nIf `parens', they look like:\n\011king@grassland.com (Elvis Parsley)\nIf `angles', they look like:\n\011Elvis Parsley <king@grassland.com>\n\nOtherwise, most addresses look like `angles', but they look like\n`parens' if `angles' would need quoting and `parens' would not." "23.2" (choice (const :tag "simple" nil) (const parens) (const angles) (const default)) message-insert-canlock #f(compiled-function () #<bytecode 0xfe59dd>) "Whether to insert a Cancel-Lock header in news postings." message-syntax-checks #f(compiled-function () #<bytecode 0xfe59e9>) "Controls what syntax checks should not be performed on outgoing posts.\nTo disable checking of long signatures, for instance, add\n `(signature . disabled)' to this list.\n\nDon't touch this variable unless you really know what you're doing.\n\nChecks include `approved', `bogus-recipient', `continuation-headers',\n`control-chars', `empty', `existing-newsgroups', `from', `illegible-text',\n`invisible-text', `long-header-lines', `long-lines', `message-id',\n`multiple-headers', `new-text', `newsgroups', `quoting-style',\n`repeated-newsgroups', `reply-to', `sender', `sendsys', `shoot',\n`shorten-followup-to', `signature', `size', `subject', `subject-cmsg'\nand `valid-newsgroups'." (repeat sexp) message-required-headers #f(compiled-function () #<bytecode 0xfe59f5>) "Headers to be generated or prompted for when sending a message.\nAlso see `message-required-news-headers' and\n`message-required-mail-headers'." (custom-manual "(message)Message Headers") (repeat sexp) message-draft-headers #f(compiled-function () #<bytecode 0xfe5a01>) "Headers to be generated when saving a draft message." (custom-manual "(message)Message Headers") (repeat sexp) message-required-news-headers #f(compiled-function () #<bytecode 0xfe5a0d>) "Headers to be generated or prompted for when posting an article.\nRFC977 and RFC1036 require From, Date, Newsgroups, Subject,\nMessage-ID.  Organization, Lines, In-Reply-To, Expires, and\nUser-Agent are optional.  If you don't want message to insert some\nheader, remove it from this list." (custom-manual "(message)Message Headers") (repeat sexp) message-required-mail-headers #f(compiled-function () #<bytecode 0xfe5a19>) "Headers to be generated or prompted for when mailing a message.\nIt is recommended that From, Date, To, Subject and Message-ID be\nincluded.  Organization and User-Agent are optional." (custom-manual "(message)Message Headers") (repeat sexp) message-prune-recipient-rules #f(compiled-function () #<bytecode 0xfe5a25>) "Rules for how to prune the list of recipients when doing wide replies.\nThis is a list of regexps and regexp matches." "24.1" (custom-manual "(message)Wide Reply") (repeat regexp) message-deletable-headers #f(compiled-function () #<bytecode 0xfe5a31>) "Headers to delete if present and previously generated by message." (custom-manual "(message)Message Headers") (repeat (symbol :tag "Header")) message-ignored-news-headers #f(compiled-function () #<bytecode 0xfe5a3d>) "Regexp of headers to be removed unconditionally before posting." (custom-manual "(message)Message Headers") (repeat :value-to-internal (lambda (widget value) (custom-split-regexp-maybe value)) :match (lambda (widget value) (or (stringp value) (widget-editable-list-match widget value))) regexp) message-ignored-mail-headers #f(compiled-function () #<bytecode 0xfe5a49>) "Regexp of headers to be removed unconditionally before mailing." (custom-manual "(message)Mail Headers") message-ignored-supersedes-headers #f(compiled-function () #<bytecode 0xfadd05>) "Header lines matching this regexp will be deleted before posting.\nIt's best to delete old Path and Date headers before posting to avoid\nany confusion." (custom-manual "(message)Superseding") (repeat :value-to-internal (lambda (widget value) (custom-split-regexp-maybe value)) :match (lambda (widget value) (or (stringp value) (widget-editable-list-match widget value))) regexp) message-subject-re-regexp #f(compiled-function () #<bytecode 0xfadd11>) "Regexp matching \"Re: \" in the subject line." (custom-manual "(message)Message Headers") message-subject-trailing-was-query #f(compiled-function () #<bytecode 0xfadd1d>) "What to do with trailing \"(was: <old subject>)\" in subject lines.\nIf nil, leave the subject unchanged.  If it is the symbol `ask', query\nthe user what to do.  In this case, the subject is matched against\n`message-subject-trailing-was-ask-regexp'.  If\n`message-subject-trailing-was-query' is t, always strip the trailing\nold subject.  In this case, `message-subject-trailing-was-regexp' is\nused." (choice (const :tag "never" nil) (const :tag "always strip" t) (const ask)) (custom-manual "(message)Message Headers") message-subject-trailing-was-ask-regexp #f(compiled-function () #<bytecode 0xfadd29>) "Regexp matching \"(was: <old subject>)\" in the subject line.\n\nThe function `message-strip-subject-trailing-was' uses this regexp if\n`message-subject-trailing-was-query' is set to the symbol `ask'.  If\nthe variable is t instead of `ask', use\n`message-subject-trailing-was-regexp' instead.\n\nIt is okay to create some false positives here, as the user is asked." (custom-manual "(message)Message Headers") message-subject-trailing-was-regexp #f(compiled-function () #<bytecode 0xfadd35>) "Regexp matching \"(was: <old subject>)\" in the subject line.\n\nIf `message-subject-trailing-was-query' is set to t, the subject is\nmatched against `message-subject-trailing-was-regexp' in\n`message-strip-subject-trailing-was'.  You should use a regexp creating very\nfew false positives here." (custom-manual "(message)Message Headers") message-mark-insert-begin #f(compiled-function () #<bytecode 0xfadd41>) "How to mark the beginning of some inserted text." string (custom-manual "(message)Insertion Variables") message-mark-insert-end #f(compiled-function () #<bytecode 0xfadd4d>) "How to mark the end of some inserted text." (custom-manual "(message)Insertion Variables") message-archive-header #f(compiled-function () #<bytecode 0xfadd59>) "Header to insert when you don't want your article to be archived.\nArchives (such as groups.google.com) respect this header." (custom-manual "(message)Header Commands") message-archive-note #f(compiled-function () #<bytecode 0xfadd65>) "Note to insert why you wouldn't want this posting archived.\nIf nil, don't insert any text in the body." (radio string (const nil)) (custom-manual "(message)Header Commands")] 14)
  require(message)
  byte-code("\300\301!\210\300\302!\210\300\303!\210\300\304!\210\305\306\307\310\311\312\311\313\314\315\316\317&\013\207" [require add-log pcvs-util ring message custom-declare-group log-edit nil "Major mode for editing RCS and CVS commit messages." :group pcl-cvs vc :version "21.1" :prefix "log-edit-"] 12)
  require(log-edit)
  byte-code("\300\301!\210\300\302!\210\300\303\304\305#\210\300\306\304\305#\210\300\307!\210\300\310!\210\300\311!\210\300\312!\210\300\313!\210\300\314!\210\315\316\304\317\320\321\322\323\324\325&\011\210\326\327\330\331\332DD\333\334\335\336\337\324\316\340\341&\013\207" [require dash subr-x magit-git nil t magit-utils log-edit ring rx server transient with-editor custom-declare-group git-commit "Edit Git commit messages." :prefix "git-commit-" :link (info-link "(magit)Editing Commit Messages") :group tools custom-declare-variable global-git-commit-mode funcall function #f(compiled-function () #<bytecode 0xfe323d>) "Non-nil if Global Git-Commit mode is enabled.\nSee the `global-git-commit-mode' command\nfor a description of this minor mode.\nSetting this variable directly does not take effect;\neither customize it (see the info node `Easy Customization')\nor call the function `global-git-commit-mode'." :set custom-set-minor-mode :initialize #f(compiled-function (symbol exp) #<bytecode 0xfe3255>) :type boolean] 12)
  require(git-commit)
  byte-code("\302\303!\210\302\304!\210\302\305!\210\302\306!\210\307\310\311\312\313\314\315\316\315\317&\011\210\320\321\322\323\324DD\325\315\310\326\327&\007\210\320\330\322\323\331DD\332\333\334\315\310\326\327&\011\210\320\335\322\323\336DD\337\333\340\315\310\326\341&\011\210\320\342\322\323\343DD\344\333\345\315\310\326\346&\011\210\320\347\322\323\350DD\351\333\352\353\354\315\310\326\327\355\356&\015\210\320\357\322\323\360DD\361\333\362\315\310\326\346&\011\210\320\363\322\323\364DD\365\315\310\366\367\326\370&\011\210\320\371\322\323\372DD\373\333\374\353\375\315\310\366\376\326\346&\015\210\377\363\201@\0\201A\0#\210\320\201B\0\322\323\201C\0\ DD\201D\0\333\201E\0\315\310\326\201F\0&\011\210\320\201G\0\322\323\201H\0\ DD\201I\0\315\310\366\201J\0\326\201K\0&\011\210\320\201L\0\322\323\201M\0\ DD\201N\0\333\201O\0\315\310\366\201P\0\326\201Q\0&\013\210\320\201R\0\322\323\201S\0\ DD\201T\0\315\310\366\376\326\346&\011\210\320\201U\0\322\323\201V\0\ DD\201W\0\315\310\326\201X\0\201Y\0\201Z\0\201[\0\201\\\0\201]\0\201^\0\201_\0\201`\0\010\257\005\201a\0\ BBBED&\007\210\320\201b\0\322\323\201c\0\ DD\201d\0\333\201e\0\315\310\326\346&\011\210\320\201f\0\322\323\201g\0\ DD\201h\0\333\201i\0\315\310\326\346&\011\210\320\201j\0\322\323\201k\0\ DD\201l\0\333\201m\0\315\310\326\201n\0&\011\210\320\201o\0\322\323\201p\0\ DD\201q\0\333\201r\0\315\316\315\310\326\346&\013\210\307\201s\0\311\201t\0\313\201u\0\315\317&\007\210\320\201v\0\322\323\201w\0\ DD\201x\0\315\201s\0\326\327\355\201y\0&\011\210\320\201z\0\322\323\201{\0\ DD\201|\0\333\201}\0\315\201s\0\326\327&\011\210\320\201~\0\322\323\201\177\0\ DD\201\200\0\333\201\201\0\315\201s\0\326\201\202\0&\011\210\320\201\203\0\322\323\201\204\0\ DD\201\205\0\333\201\206\0\315\201s\0\326\201\207\0&\011\210\320\201\210\0\322\323\201\211\0\ DD\201\212\0\333\201\213\0\315\201s\0\326\201\214\0&\011\210\320\201\215\0\322\323\201\216\0\ DD\201\217\0\333\201\220\0\315\201s\0\326\201\221\0&\011\210\320\201\222\0\322\323\201\223\0\ DD\201\224\0\333\201\225\0\315\201s\0\326\346&\011\210\320\201\226\0\322\323\201\227\0\ DD\201\230\0\333\201\231\0\315\201s\0\326\201\232\0&\011\210\320\201\233\0\322\323\201\234\0\ DD\201\235\0\333\201\236\0\315\201s\0\326\346&\011\210\320\201\237\0\322\323\201\240\0\ DD\201\241\0\333\201\242\0\315\310\326\346&\011\210\320\201\243\0\322\323\201\244\0\ DD\201\245\0\333\201\246\0\315\310\326\346&\011\210\201\247\0\201\250\0\201A\0\201\251\0\011\201\252\0Y\205\032\003\201\253\0\201\254\0\"BC\201\255\0\315\201\256\0%\210\201\247\0\201\257\0\201A\0\201\251\0\011\201\252\0Y\205@\003\201\260\0\201\261\0\"BC\201\262\0\315\201\256\0%\210\201\247\0\201\263\0\201\264\0\201\251\0\011\201\252\0Y\205f\003\201\265\0\201\266\0\"B\201\267\0\201\251\0\011\201\252\0Y\205|\003\201\270\0\201\271\0\"BD\201\272\0\315\201\256\0%\210\201\247\0\201\273\0\201\274\0\201\251\0\011\201\252\0Y\205\242\003\201\275\0\201\276\0\"B\201\277\0\201\251\0\011\201\252\0Y\205\270\003\201\300\0\201\301\0\"BD\201\302\0\315\201\256\0%\210\201\247\0\201\303\0\201\304\0\201\251\0\011\201\252\0Y\205\336\003\201\305\0\201\306\0\"B\201\307\0\201\251\0\011\201\252\0Y\205\364\003\201\310\0\201\311\0\"BD\201\312\0\315\201\256\0%\210\201\247\0\201\313\0\201\314\0\201\251\0\011\201\252\0Y\205\032\004\201\315\0\201\316\0\"B\201\317\0\201\251\0\011\201\252\0Y\2050\004\201\320\0\201\321\0\"BD\201\322\0\315\201\256\0%\210\201\247\0\201\323\0\201A\0\201\324\0\201\325\0\011\201\252\0Y\205p\004\201\326\0\201\327\0\ 1m\004\201\330\0\201\331\0\201\326\0\"0\202o\004\210\311DBBBC\201\332\0\315\201\256\0%\210\201\247\0\201\333\0\201\334\0\201\335\0\315\201\256\0%\210\201\247\0\201\336\0\201\337\0\201\340\0\315\201\256\0%\210\201\247\0\201\341\0\201\342\0\201\251\0\011\201\252\0Y\205\270\004\201\343\0\201\344\0\"B\201\345\0\201\251\0\011\201\252\0Y\205\316\004\201\346\0\201\347\0\"BD\201\350\0\315\201\256\0%\210\201\247\0\201\351\0\201A\0\201\251\0\011\201\252\0Y\205\364\004\201\352\0\201\353\0\"BC\201\354\0\315\201\256\0%\210\201\247\0\201\355\0\201\356\0\201\357\0\315\201\256\0%\210\201\247\0\201\360\0\201\361\0\201\251\0\011\201\252\0Y\205,\005\201\362\0\201\363\0\"B\201\364\0\201\251\0\011\201\252\0Y\205B\005\201\365\0\201\366\0\"BD\201\367\0\315\201\256\0%\210\201\247\0\201\370\0\201\371\0\201\251\0\011\201\252\0Y\205h\005\201\372\0\201\373\0\"B\201\374\0\201\251\0\011\201\252\0Y\205~\005\201\375\0\201\376\0\"BD\201\377\0\315\201\256\0%\210\201\247\0\201\0\001\201\001\001\201\002\001\315\201\256\0%\210\201\247\0\201\003\001\201\004\001\201\251\0\011\201\252\0Y\205\266\005\201\005\001\201\006\001\"B\201\007\001\201\251\0\011\201\252\0Y\205\314\005\201\010\001\201\011\001\"BD\201\n\001\315\201\256\0%\210\201\247\0\201\013\001\201\f\001\201\015\001\315\201\256\0%\210\201\247\0\201\016\001\201\017\001\201\251\0\011\201\252\0Y\205\004\006\201\020\001\201\021\001\"B\201\022\001\201\251\0\011\201\252\0Y\205\032\006\201\023\001\201\024\001\"BD\201\025\001\315\201\256\0%\210\201\247\0\201\026\001\201\027\001\201\251\0\011\201\252\0Y\205@\006\201\030\001\201\031\001\"B\201\032\001\201\251\0\011\201\252\0Y\205V\006\201\033\001\201\034\001\"BD\201\367\0\315\201\256\0%\210\201\247\0\201\035\001\201\036\001\201\251\0\011\201\252\0Y\205|\006\201\037\001\201 \001\"B\201!\001\201\251\0\011\201\252\0Y\205\222\006\201\"\001\201#\001\"BD\201\377\0\315\201\256\0%\210\201\247\0\201$\001\201%\001\201\002\001\315\201\256\0%\210\201\247\0\201&\001\201'\001\201\251\0\011\201\252\0Y\205\312\006\201(\001\201)\001\"B\201*\001\201\251\0\011\201\252\0Y\205\340\006\201+\001\201,\001\"BD\201\n\001\315\201\256\0%\210\201\247\0\201-\001\201.\001\201\015\001\315\201\256\0%\210\201\247\0\201/\001\2010\001\201\251\0\011\201\252\0Y\205\030\007\2011\001\2012\001\"B\2013\001\201\251\0\011\201\252\0Y\205.\007\2014\001\2015\001\"BD\2016\001\315\201\256\0%\210\201\247\0\2017\001\2018\001\2019\001\315\201\256\0%\210\201\247\0\201:\001\201;\001\201<\001\315\201\256\0%\210\201\247\0\201=\001\201>\001\201?\001\315\201\256\0%\210\201@\001\201A\001\201B\001\201C\001!\"\210\201@\001\201D\001\201E\001\201C\001!\"\210\201@\001\201F\001\201D\001\"\210\201G\001\201F\001\201H\001\201I\001#\210\201J\001\201C\001\201K\001\201D\001#\201L\001\201C\001\201M\001\201N\001\311$\207" [tab-width emacs-major-version require magit-core git-commit diff-mode smerge-mode custom-declare-group magit-diff nil "Inspect and manipulate Git diffs." :link (info-link "(magit)Diffing") :group magit-commands magit-modes custom-declare-variable magit-diff-mode-hook funcall function #f(compiled-function () #<bytecode 0xe1a2e9>) "Hook run after entering Magit-Diff mode." :type hook magit-diff-sections-hook #f(compiled-function () #<bytecode 0xfdfa19>) "Hook run to insert sections into a `magit-diff-mode' buffer." :package-version (magit . "2.3.0") magit-diff-expansion-threshold #f(compiled-function () #<bytecode 0xfdfa25>) "After how many seconds not to expand anymore diffs.\n\nExcept in status buffers, diffs are usually start out fully\nexpanded.  Because that can take a long time, all diffs that\nhaven't been fontified during a refresh before the threshold\ndefined here are instead displayed with their bodies collapsed.\n\nNote that this can cause sections that were previously expanded\nto be collapsed.  So you should not pick a very low value here.\n\nThe hook function `magit-diff-expansion-threshold' has to be a\nmember of `magit-section-set-visibility-hook' for this option\nto have any effect." (magit . "2.9.0") float magit-diff-highlight-hunk-body #f(compiled-function () #<bytecode 0xfdfa31>) "Whether to highlight bodies of selected hunk sections.\nThis only has an effect if `magit-diff-highlight' is a\nmember of `magit-section-highlight-hook', which see." (magit . "2.1.0") boolean magit-diff-highlight-hunk-region-functions #f(compiled-function () #<bytecode 0xfe3305>) "The functions used to highlight the hunk-internal region.\n\n`magit-diff-highlight-hunk-region-dim-outside' overlays the outside\nof the hunk internal selection with a face that causes the added and\nremoved lines to have the same background color as context lines.\nThis function should not be removed from the value of this option.\n\n`magit-diff-highlight-hunk-region-using-overlays' and\n`magit-diff-highlight-hunk-region-using-underline' emphasize the\nregion by placing delimiting horizontal lines before and after it.\nThe underline variant was implemented because Eli said that is\nhow we should do it.  However the overlay variant actually works\nbetter.  Also see https://github.com/magit/magit/issues/2758.\n\nInstead of, or in addition to, using delimiting horizontal lines,\nto emphasize the boundaries, you may which to emphasize the text\nitself, using `magit-diff-highlight-hunk-region-using-face'.\n\nIn terminal frames it's not possible to draw lines as the overlay\nand underline variants normally do, so there they fall back to\ncalling the face function instead." (magit . "2.9.0") :set-after (magit-diff-show-lines-boundaries) :options (magit-diff-highlight-hunk-region-dim-outside magit-diff-highlight-hunk-region-using-underline magit-diff-highlight-hunk-region-using-overlays magit-diff-highlight-hunk-region-using-face) magit-diff-unmarked-lines-keep-foreground #f(compiled-function () #<bytecode 0xfe3311>) "Whether `magit-diff-highlight-hunk-region-dim-outside' preserves foreground.\nWhen this is set to nil, then that function only adjusts the\nforeground color but added and removed lines outside the region\nkeep their distinct foreground colors." (magit . "2.9.0") magit-diff-refine-hunk #f(compiled-function () #<bytecode 0xfe331d>) "Whether to show word-granularity differences within diff hunks.\n\nnil    Never show fine differences.\nt      Show fine differences for the current diff hunk only.\n`all'  Show fine differences for all displayed diff hunks." :safe #f(compiled-function (val) #<bytecode 0xfe8841>) (choice (const :tag "Never" nil) (const :tag "Current" t) (const :tag "All" all)) magit-diff-refine-ignore-whitespace #f(compiled-function () #<bytecode 0xfe884d>) "Whether to ignore whitespace changes in word-granularity differences." (magit . "3.0.0") (smerge-refine-ignore-whitespace) booleanp put permanent-local t magit-diff-adjust-tab-width #f(compiled-function () #<bytecode 0xfe8859>) "Whether to adjust the width of tabs in diffs.\n\nDetermining the correct width can be expensive if it requires\nopening large and/or many files, so the widths are cached in\nthe variable `magit-diff--tab-width-cache'.  Set that to nil\nto invalidate the cache.\n\nnil       Never adjust tab width.  Use `tab-width's value from\n          the Magit buffer itself instead.\n\nt         If the corresponding file-visiting buffer exits, then\n          use `tab-width's value from that buffer.  Doing this is\n          cheap, so this value is used even if a corresponding\n          cache entry exists.\n\n`always'  If there is no such buffer, then temporarily visit the\n          file to determine the value.\n\nNUMBER    Like `always', but don't visit files larger than NUMBER\n          bytes." (magit . "2.12.0") (choice (const :tag "Never" nil) (const :tag "If file-visiting buffer exists" t) (integer :tag "If file isn't larger than N bytes") (const :tag "Always" always)) magit-diff-paint-whitespace #f(compiled-function () #<bytecode 0xfe0d65>) "Specify where to highlight whitespace errors.\n\nnil            Never highlight whitespace errors.\nt              Highlight whitespace errors everywhere.\n`uncommitted'  Only highlight whitespace errors in diffs\n               showing uncommitted changes.\n\nFor backward compatibility `status' is treated as a synonym\nfor `uncommitted'.\n\nThe option `magit-diff-paint-whitespace-lines' controls for\nwhat lines (added/remove/context) errors are highlighted.\n\nThe options `magit-diff-highlight-trailing' and\n`magit-diff-highlight-indentation' control what kind of\nwhitespace errors are highlighted." #f(compiled-function (val) #<bytecode 0xfe0d71>) (choice (const :tag "In all diffs" t) (const :tag "Only in uncommitted changes" uncommitted) (const :tag "Never" nil)) magit-diff-paint-whitespace-lines #f(compiled-function () #<bytecode 0xfe0d7d>) "Specify in what kind of lines to highlight whitespace errors.\n\nt         Highlight only in added lines.\n`both'    Highlight in added and removed lines.\n`all'     Highlight in added, removed and context lines." (magit . "3.0.0") #f(compiled-function (val) #<bytecode 0xfe2ced>) (choice (const :tag "in added lines" t) (const :tag "in added and removed lines" both) (const :tag "in added, removed and context lines" all)) magit-diff-highlight-trailing #f(compiled-function () #<bytecode 0xfe2cf9>) "Whether to highlight whitespace at the end of a line in diffs.\nUsed only when `magit-diff-paint-whitespace' is non-nil." magit-diff-highlight-indentation #f(compiled-function () #<bytecode 0xfe2d05>) "Highlight the \"wrong\" indentation style.\nUsed only when `magit-diff-paint-whitespace' is non-nil.\n\nThe value is an alist of the form ((REGEXP . INDENT)...).  The\npath to the current repository is matched against each element\nin reverse order.  Therefore if a REGEXP matches, then earlier\nelements are not tried.\n\nIf the used INDENT is `tabs', highlight indentation with tabs.\nIf INDENT is an integer, highlight indentation with at least\nthat many spaces.  Otherwise, highlight neither." repeat cons (string :tag "Directory regexp") choice (const :tag "Tabs" tabs) integer :tag "Spaces" :value ((const :tag "Neither" nil)) magit-diff-hide-trailing-cr-characters #f(compiled-function () #<bytecode 0xfe4aa5>) "Whether to hide ^M characters at the end of a line in diffs." (magit . "2.6.0") magit-diff-highlight-keywords #f(compiled-function () #<bytecode 0xfe4ab1>) "Whether to highlight bracketed keywords in commit messages." (magit . "2.12.0") magit-diff-extra-stat-arguments #f(compiled-function () #<bytecode 0xfe4abd>) "Additional arguments to be used alongside `--stat'.\n\nA list of zero or more arguments or a function that takes no\nargument and returns such a list.  These arguments are allowed\nhere: `--stat-width', `--stat-name-width', `--stat-graph-width'\nand `--compact-summary'.  See the git-diff(1) manpage." (magit . "3.0.0") (radio (function-item magit-diff-use-window-width-as-stat-width) function (list string) (const :tag "None" nil)) magit-diff-buffer-file-locked #f(compiled-function () #<bytecode 0xfe49f9>) "Whether `magit-diff-buffer-file' uses a dedicated buffer." (magit . "2.7.0") magit-revision "Inspect and manipulate Git commits." (info-link "(magit)Revision Buffer") magit-revision-mode-hook #f(compiled-function () #<bytecode 0xfe4a05>) "Hook run after entering Magit-Revision mode." (bug-reference-mode goto-address-mode) magit-revision-sections-hook #f(compiled-function () #<bytecode 0xfe4a11>) "Hook run to insert sections into a `magit-revision-mode' buffer." (magit . "2.3.0") magit-revision-headers-format #f(compiled-function () #<bytecode 0xfe9095>) "Format string used to insert headers in revision buffers.\n\nAll headers in revision buffers are inserted by the section\ninserter `magit-insert-revision-headers'.  Some of the headers\nare created by calling `git show --format=FORMAT' where FORMAT\nis the format specified here.  Other headers are hard coded or\nsubject to option `magit-revision-insert-related-refs'." (magit . "2.3.0") string magit-revision-insert-related-refs #f(compiled-function () #<bytecode 0xfe90a1>) "Whether to show related branches in revision buffers\n\n`nil'   Don't show any related branches.\n`t'     Show related local branches.\n`all'   Show related local and remote branches.\n`mixed' Show all containing branches and local merged branches." (magit . "2.1.0") (choice (const :tag "don't" nil) (const :tag "local only" t) (const :tag "all related" all) (const :tag "all containing, local merged" mixed)) magit-revision-use-hash-sections #f(compiled-function () #<bytecode 0xfe90ad>) "Whether to turn hashes inside the commit message into sections.\n\nIf non-nil, then hashes inside the commit message are turned into\n`commit' sections.  There is a trade off to be made between\nperformance and reliability:\n\n- `slow' calls git for every word to be absolutely sure.\n- `quick' skips words less than seven characters long.\n- `quicker' additionally skips words that don't contain a number.\n- `quickest' uses all words that are at least seven characters\n  long and which contain at least one number as well as at least\n  one letter.\n\nIf nil, then no hashes are turned into sections, but you can\nstill visit the commit at point using \"RET\"." (magit . "2.12.0") (choice (const :tag "Use sections, quickest" quickest) (const :tag "Use sections, quicker" quicker) (const :tag "Use sections, quick" quick) (const :tag "Use sections, slow" slow) (const :tag "Don't use sections" nil)) magit-revision-show-gravatars #f(compiled-function () #<bytecode 0xfe8f8d>) "Whether to show gravatar images in revision buffers.\n\nIf nil, then don't insert any gravatar images.  If t, then insert\nboth images.  If `author' or `committer', then insert only the\nrespective image.\n\nIf you have customized the option `magit-revision-header-format'\nand want to insert the images then you might also have to specify\nwhere to do so.  In that case the value has to be a cons-cell of\ntwo regular expressions.  The car specifies where to insert the\nauthor's image.  The top half of the image is inserted right\nafter the matched text, the bottom half on the next line in the\nsame column.  The cdr specifies where to insert the committer's\nimage, accordingly.  Either the car or the cdr may be nil." (magit . "2.3.0") (choice (const :tag "Don't show gravatars" nil) (const :tag "Show gravatars" t) (const :tag "Show author gravatar" author) (const :tag "Show committer gravatar" committer) (cons :tag "Show gravatars using custom pattern." (regexp :tag "Author regexp" "^Author:     ") (regexp :tag "Committer regexp" "^Commit:     "))) magit-revision-use-gravatar-kludge #f(compiled-function () #<bytecode 0xfe8f99>) "Whether to work around a bug which affects display of gravatars.\n\nGravatar images are spliced into two halves which are then\ndisplayed on separate lines.  On OS X the splicing has a bug in\nsome Emacs builds, which causes the top and bottom halves to be\ninterchanged.  Enabling this option works around this issue by\ninterchanging the halves once more, which cancels out the effect\nof the bug.\n\nSee https://github.com/magit/magit/issues/2265\nand https://debbugs.gnu.org/cgi/bugreport.cgi?bug=7847.\n\nStarting with Emacs 26.1 this kludge should not be required for\nany build." (magit . "2.3.0") magit-revision-fill-summary-line #f(compiled-function () #<bytecode 0xfe8fa5>) "Whether to fill excessively long summary lines.\n\nIf this is an integer, then the summary line is filled if it is\nlonger than either the limit specified here or `window-width'.\n\nYou may want to only set this locally in \".dir-locals-2.el\" for\nrepositories known to contain bad commit messages.\n\nThe body of the message is left alone because (a) most people who\nwrite excessively long summary lines usually don't add a body and\n(b) even people who have the decency to wrap their lines may have\na good reason to include a long line in the body sometimes." (magit . "2.90.0") (choice (const :tag "Don't fill" nil) (integer :tag "Fill if longer than")) magit-revision-filter-files-on-follow #f(compiled-function () #<bytecode 0xfea525>) "Whether to honor file filter if log arguments include --follow.\n\nWhen a commit is displayed from a log buffer, the resulting\nrevision buffer usually shares the log's file arguments,\nrestricting the diff to those files.  However, there's a\ncomplication when the log arguments include --follow: if the log\nfollows a file across a rename event, keeping the file\nrestriction would mean showing an empty diff in revision buffers\nfor commits before the rename event.\n\nWhen this option is nil, the revision buffer ignores the log's\nfilter if the log arguments include --follow.  If non-nil, the\nlog's file filter is always honored." (magit . "3.0.0") magit-diff-visit-previous-blob #f(compiled-function () #<bytecode 0xfea531>) "Whether `magit-diff-visit-file' may visit the previous blob.\n\nWhen this is t and point is on a removed line in a diff for a\ncommitted change, then `magit-diff-visit-file' visits the blob\nfrom the last revision which still had that line.\n\nCurrently this is only supported for committed changes, for\nstaged and unstaged changes `magit-diff-visit-file' always\nvisits the file in the working tree." (magit . "2.9.0") magit-diff-visit-avoid-head-blob #f(compiled-function () #<bytecode 0xfea53d>) "Whether `magit-diff-visit-file' avoids visiting a blob from `HEAD'.\n\nBy default `magit-diff-visit-file' always visits the blob that\nadded the current line, while `magit-diff-visit-worktree-file'\nvisits the respective file in the working tree.  For the `HEAD'\ncommit, the former command used to visit the worktree file too,\nbut that made it impossible to visit a blob from `HEAD'.\n\nWhen point is on a removed line and that change has not been\ncommitted yet, then `magit-diff-visit-file' now visits the last\nblob that still had that line, which is a blob from `HEAD'.\nPreviously this function used to visit the worktree file not\nonly for added lines but also for such removed lines.\n\nIf you prefer the old behaviors, then set this to t." (magit . "3.0.0") custom-declare-face magit-diff-file-heading append 27 (:extend t) (:weight bold) "Face for diff file headings." magit-faces magit-diff-file-heading-highlight (:extend t) (:inherit magit-section-highlight) "Face for current diff file headings." magit-diff-file-heading-selection ((class color) (background light)) (:extend t) (:inherit magit-diff-file-heading-highlight :foreground "salmon4") ((class color) (background dark)) (:extend t) (:inherit magit-diff-file-heading-highlight :foreground "LightSalmon3") "Face for selected diff file headings." magit-diff-hunk-heading ((class color) (background light)) (:extend t) (:background "grey80" :foreground "grey30") ((class color) (background dark)) (:extend t) (:background "grey25" :foreground "grey70") "Face for diff hunk headings." magit-diff-hunk-heading-highlight ((class color) (background light)) (:extend t) (:background "grey75" :foreground "grey30") ((class color) (background dark)) (:extend t) (:background "grey35" :foreground "grey70") "Face for current diff hunk headings." magit-diff-hunk-heading-selection ((class color) (background light)) (:extend t) (:inherit magit-diff-hunk-heading-highlight :foreground "salmon4") ((class color) (background dark)) (:extend t) (:inherit magit-diff-hunk-heading-highlight :foreground "LightSalmon3") "Face for selected diff hunk headings." magit-diff-hunk-region :inherit bold :extend (error) face-attribute region "Face used by `magit-diff-highlight-hunk-region-using-face'.\n\nThis face is overlaid over text that uses other hunk faces,\nand those normally set the foreground and background colors.\nThe `:foreground' and especially the `:background' properties\nshould be avoided here.  Setting the latter would cause the\nloss of information.  Good properties to set here are `:weight'\nand `:slant'." magit-diff-revision-summary ((t :inherit magit-diff-hunk-heading)) "Face for commit message summaries." magit-diff-revision-summary-highlight ((t :inherit magit-diff-hunk-heading-highlight)) "Face for highlighted commit message summaries." magit-diff-lines-heading ((class color) (background light)) (:extend t) (:inherit magit-diff-hunk-heading-highlight :background "LightSalmon3") ((class color) (background dark)) (:extend t) (:inherit magit-diff-hunk-heading-highlight :foreground "grey80" :background "salmon4") "Face for diff hunk heading when lines are marked." magit-diff-lines-boundary (:extend t) (:inherit magit-diff-lines-heading) "Face for boundary of marked lines in diff hunk." magit-diff-conflict-heading ((t :inherit magit-diff-hunk-heading)) "Face for conflict markers." magit-diff-added ((class color) (background light)) (:extend t) (:background "#ddffdd" :foreground "#22aa22") ((class color) (background dark)) (:extend t) (:background "#335533" :foreground "#ddffdd") "Face for lines in a diff that have been added." magit-diff-removed ((class color) (background light)) (:extend t) (:background "#ffdddd" :foreground "#aa2222") ((class color) (background dark)) (:extend t) (:background "#553333" :foreground "#ffdddd") "Face for lines in a diff that have been removed." magit-diff-our ((t :inherit magit-diff-removed)) "Face for lines in a diff for our side in a conflict." magit-diff-base ((class color) (background light)) (:extend t) (:background "#ffffcc" :foreground "#aaaa11") ((class color) (background dark)) (:extend t) (:background "#555522" :foreground "#ffffcc") "Face for lines in a diff for the base side in a conflict." magit-diff-their ((t :inherit magit-diff-added)) "Face for lines in a diff for their side in a conflict." magit-diff-context ((class color) (background light)) (:extend t) (:foreground "grey50") ((class color) (background dark)) (:extend t) (:foreground "grey70") "Face for lines in a diff that are unchanged." magit-diff-added-highlight ((class color) (background light)) (:extend t) (:background "#cceecc" :foreground "#22aa22") ((class color) (background dark)) (:extend t) (:background "#336633" :foreground "#cceecc") magit-diff-removed-highlight ((class color) (background light)) (:extend t) (:background "#eecccc" :foreground "#aa2222") ((class color) (background dark)) (:extend t) (:background "#663333" :foreground "#eecccc") magit-diff-our-highlight ((t :inherit magit-diff-removed-highlight)) magit-diff-base-highlight ((class color) (background light)) (:extend t) (:background "#eeeebb" :foreground "#aaaa11") ((class color) (background dark)) (:extend t) (:background "#666622" :foreground "#eeeebb") magit-diff-their-highlight ((t :inherit magit-diff-added-highlight)) magit-diff-context-highlight ((class color) (background light)) (:extend t) (:background "grey95" :foreground "grey50") ((class color) (background dark)) (:extend t) (:background "grey20" :foreground "grey70") "Face for lines in the current context in a diff." magit-diff-whitespace-warning ((t :inherit trailing-whitespace)) "Face for highlighting whitespace errors added lines." magit-diffstat-added ((((class color) (background light)) :foreground "#22aa22") (((class color) (background dark)) :foreground "#448844")) "Face for plus sign in diffstat." magit-diffstat-removed ((((class color) (background light)) :foreground "#aa2222") (((class color) (background dark)) :foreground "#aa4444")) "Face for minus sign in diffstat." defalias magit-diff-prefix-p eieio-make-class-predicate magit-diff-prefix magit-diff-prefix--eieio-childp eieio-make-child-predicate magit-diff-prefix-child-p make-obsolete "use (cl-typep ... \\='magit-diff-prefix) instead" "25.1" define-symbol-prop cl-deftype-satisfies eieio-defclass-internal (transient-prefix) ((history-key :initform 'magit-diff) (major-mode :initform 'magit-diff-mode))] 17)
  require(magit-diff)
  byte-code("\300\301!\210\300\302!\210\300\303!\210\300\304!\210\300\305!\210\300\306!\210\300\307!\210\300\310!\210\300\311\312\313#\210\300\314!\207" [require magit-core magit-diff magit-log magit-wip magit-apply magit-repos git-commit format-spec package nil t with-editor] 4)
  load("/home/nr/.emacs.d/elpa/magit-3.1.1/magit" nil t)
  command-line-1(("--eval" "(setq debug-on-error t)" "-L" "/home/nr/.emacs.d/elpa/dash-20210609.1330/" "-L" "/home/nr/.emacs.d/elpa/transient-0.3.6/" "-L" "/home/nr/.emacs.d/elpa/with-editor-20210524.1654/" "-L" "/home/nr/.emacs.d/elpa/magit-3.1.1/" "-L" "/home/nr/.emacs.d/elpa/magit-section-3.1.1/" "-L" "/home/nr/.emacs.d/elpa/git-commit-20210525.844/" "-l" "/home/nr/.emacs.d/elpa/magit-3.1.1/magit"))
  command-line()
  normal-top-level()
```

To confirm that the issue is still present in Magit 3.1.1, I have used command

````
emacs --no-init-file -l ~/.emacs.d/magit-only.el
````

where the contents of `magit-only.el` are as follows:

```
(package-initialize)
(require 'magit)
```

Emacs launches with no errors, `C-x g` brings up Magit as expected, and typing `c c` with a file staged in the index results in the commit edit message being opened in a new frame.

**Comment by tarsius** [comment]
Somehow loading `git-commit` ends up loading `rmail`. It does do that for me too. I don't think it should do that but that appears to be an issue in Emacs, not Magit. On your system that then causes somewhere in `rmail`. That issue has very little to do with Magit, so please investigate it yourself.

**Comment by phil-s** [comment]
It might not be only path, but I can see:

git-commit -> logedit -> message -> mml -> mml-sec -> gnus-util -> rmail

**Comment by tarsius** [comment]
Yes I say the path too, but feel like somewhere something should stop loading and delay doing so until actual usage by the user makes it impossible to further delay doing so. I am guessing that somewhere a setting that controls which email package the user wants to use is unset and so we fallback to `rmail`. But maybe I am wrong about this and either this cannot be avoided or maybe it is actually the right thing to do. But that is not the point here.

The point is that there appears to be a bug in `rmail` and I don't want to be the one that figures out what that bug is.

The problem might be that it fails if "/usr/bin/movemail" exists but that it doesn't on the users machine. And maybe the authors of `rmail` felt it was okay to depend on that and blow up if it doesn't exist because "nobody" would load `rmail` if they didn't configure it and wanted to use it. That's why I mentioned that maybe `rmail` should not be loaded. All this obviously is only speculation.

All I know is that it appears that so far only one person is affected (and only while debugging) and that Magit does not appear to be the cause. So I don't want to investigate it.

----

Sadly that makes it harder to investigate the other problem. Here too so far only one person seems to be affected. I don't know how I can help with this as it sounds "impossible". Without loading third-party code or configuration this cannot happen, the user indicates that no such thing is being loaded, and yet it happens.

**Comment by phil-s** [comment]
Well the trigger in `rmail.el` is this:

```el
(defun rmail-movemail-variant-p (&rest variants)
  "Return t if the current movemail variant is any of VARIANTS.
Currently known variants are `emacs' and `mailutils'."
  (when (not rmail-movemail-variant-in-use)
    ;; Autodetect
    (setq rmail-movemail-variant-in-use (rmail-autodetect)))
  (not (null (member rmail-movemail-variant-in-use variants))))

;; Call for effect, to set rmail-movemail-program (if not set by the
;; user), and rmail-movemail-variant-in-use. Used by various functions.
;; I'm not sure if M-x rmail is the only entry point to this package.
;; If so, this can be moved there.
(rmail-movemail-variant-p)
```

With the conditional variable being:

```el
(defvar rmail-movemail-variant-in-use nil
  "The movemail variant currently in use.  Known variants are:

  `emacs'     Means any implementation, compatible with the native Emacs one.
              This is the default;
  `mailutils' Means GNU mailutils implementation, capable of handling full
mail URLs as the source mailbox.")
```

So a workaround is to set one of those values for `rmail-movemail-variant-in-use` in your init file prior to `(require 'magit)`

Possibly also set this:

```el
(defcustom rmail-movemail-program nil
  "If non-nil, the file name of the `movemail' program."
  :group 'rmail-retrieve
  :type '(choice (const nil) string))
```

On my system I see values of `emacs` and `/usr/local/libexec/emacs/27.2/x86_64-pc-linux-gnu/movemail` respectively, with that latter executable having been compiled and installed as part of Emacs.

**Comment by phil-s** [comment]
Actually, I can't figure out why the error would be signalled at all:

```
Debugger entered--Lisp error: (wrong-type-argument stringp nil)
  call-process("/usr/bin/movemail" nil #<buffer  *temp*> nil "--version")
```

There's nothing wrong with the `nil` values in that `call-process` invocation -- they're both optional arguments.

**Comment by phil-s** [comment]
Well, I guess it's actually failing in C code somewhere.

It *might* be right away in the handling of the null device in `call-process` when the `infile` argument is `nil`.

```c
    infile = build_string (NULL_DEVICE);
  encoded_infile = ENCODE_FILE (infile);
  filefd = emacs_open (SSDATA (encoded_infile), O_RDONLY, 0);
```

but I suppose it could be anywhere beyond that... one would probably need to use `gdb` and put a breakpoint in `wrong_type_argument` to get a stack trace?  (I rarely look at the C code, so I'm just guessing.)

**Comment by nrnrnr** [comment]
I worry that the startup is a red herring that distracts from the original issue.

Would you be willing to try to diagnose the Magit issue based on the following emacs startup command?

    emacs -Q --eval="(progn (package-initialize) (require 'magit))"

Using that command I am able to duplicate the problem without trying to involve rmail.

**Comment by nrnrnr** [comment]
Here is another command that enables me to duplicate the problem (like the `magit-emacs-Q-command` except the last option evaluates `require` instead of loading `magit` directly):

    /usr/bin/emacs -Q --eval \(setq\ debug-on-error\ t\) -L /home/nr/.emacs.d/elpa/dash-20210609.1330/ -L /home/nr/.emacs.d/elpa/transient-0.3.6/ -L /home/nr/.emacs.d/elpa/with-editor-20210524.1654/ -L /home/nr/.emacs.d/elpa/magit-3.1.1/ -L /home/nr/.emacs.d/elpa/magit-section-3.1.1/ -L /home/nr/.emacs.d/elpa/git-commit-20210525.844/ --eval="(require 'magit)"

**Comment by tarsius** [comment]
> Would you be willing to try to diagnose the Magit issue based on the following emacs startup command?

Yes, I can do that. But I am not able to reproduce the issue when doing that (or when using some other method of loading just Magit without my configuration or unrelated pacakges).

**Comment by tarsius** [comment]
I haven't tried with Emacs 26.1 though and instead of me doing that you might want to try a more recent version.

**Comment by nrnrnr** [comment]
Got it!  I'm stuck on Debian stable, so I'm not sure how easily I'll be able to turn up an Emacs 27, but I'll look into it.

**Comment by nrnrnr** [comment]
I have duplicated the issue in Emacs 27.1:

````
Magit 3.1.1, Git 2.20.1, Emacs 27.1, gnu/linux
````

I have also confirmed that with Emacs 27.1, the following `magit-emacs-Q-command` works as expected:

````
/usr/bin/emacs -Q --eval \(setq\ debug-on-error\ t\) -L /home/nr/.emacs.d/elpa/dash-20210609.1330/ -L /home/nr/.emacs.d/elpa/transient-0.3.6/ -L /home/nr/.emacs.d/elpa/with-editor-20210524.1654/ -L /home/nr/.emacs.d/elpa/magit-3.1.1/ -L /home/nr/.emacs.d/elpa/magit-section-3.1.1/ -L /home/nr/.emacs.d/elpa/git-commit-20210525.844/ -l /home/nr/.emacs.d/elpa/magit-3.1.1/magit
````

Starting Emacs with this command also duplicates the issue.

**Comment by nrnrnr** [comment]
This issue was caused by a wrapper around `emacsclient` in the file system.  The wrapper always added the `-c` option, whether called for or not.  

More (but nothing terribly enlightening) at https://github.com/magit/magit/discussions/4443
